### DISCUSSION TITLE
Change preview

### DISCUSSION THREAD

**Discussion by og900aero** [sxyazi/yazi#584] [1. Q&A] [answered]
I've been using the preview plugin for a while now, for example, to display the properties of audio files. It's not that hard. However, I would like to change the basic preview program, which gives insight into the display of codes and text files, because its coloring is not as good as that of the batcat (Debian) program. In addition, I want to keep it so that the preview can be scrolled. How can I program this? Because the fallback command is the file command, so I cannot rewrite it.

**Comment by sxyazi** [comment]
Yazi and bat use the same syntect highlighting engine, so there is no reason for Yazi to perform any worse than bat. Have you set up a theme for syntect using `syntect_theme`? See the docs here https://yazi-rs.github.io/docs/configuration/theme/#manager

**Reply by og900aero** [reply]
No, I not use any themes from https://github.com/yazi-rs/themes. I only use batcat from Debian original repo, and use batcat own config file:

```
# This is `bat`s configuration file. Each line either contains a comment or
# a command-line option that you want to pass to `bat` by default. You can
# run `bat --help` to get a list of all possible configuration options.

# Specify desired highlighting theme (e.g. "TwoDark"). Run `bat --list-themes`
# for a list of all available themes
--theme="Dracula"

# Syntax mappings: map a certain filename pattern to a language.
--map-syntax "*.txt:Python"
```

So I can't somehow tell Yazi to use the factory theme (Debian repos's bat) built into the batcat program?

**Reply by sxyazi** [reply]
No. I mean the theme of [syntect](https://crates.io/crates/syntect), not Yazi's

- syntect_theme (String): Theme file path. Since Yazi and `bat` use the same highlighter, so you can directly use bat's theme files, such as `~/.config/bat/themes/Catppuccino-mocha.tmTheme`.

**Reply by og900aero** [reply]
The problem is, the bat have builtin themes like Dracula too, so I cant set path to the external syntect theme, because only have a config file: -/.config/bat/config

**Reply by sxyazi** [reply, marked as answer]
`syntect_theme` is in Yazi's `theme.toml`, not in bat's `~/.config/bat/config`, see

![1](https://github.com/sxyazi/yazi/assets/17523360/c14d1d66-d8b8-452f-87e9-c24fc83f8616)

**Reply by og900aero** [reply]
Sorry I misunderstood. Now working great.
